lib/api/apiUtils/object/checkUserMetadataSize.js

```diff
@@ -0,0 +1,38 @@
+const { maximumMetaHeadersSize,
+    invalidObjectUserMetadataHeader } = require('../../../../constants');
+
+/**
+ * Checks the size of the user metadata in the object metadata and removes
+ * them from the response if the size of the user metadata is larger than
+ * the maximum size allowed. A custome metadata key is added to the response
+ * with the number of user metadata keys not returned as its value
+ * @param {object} objMD - object's metadata
+ * @return {object} responseMetaHeaders headers with object metadata to include
+ * in response to client
+ */
+function checkUserMetadataSize(objMD) {
+    let md = {};
+    let userMetadataSize = 0;
+    md = JSON.parse(JSON.stringify(objMD));
```

Comment:
I think this can be simplified a bit using the public methods exposed in `objMD`. For example, it allows retrieval of user metadata fields with `getUserMetadata` or the whole md simply with `getValue`.